### PR TITLE
Add Find Restores Last Search setting for find-bar recovery

### PR DESCRIPTION
**Pull request by prncvrm**
## Summary

Reopening find (Cmd+F) in a terminal or browser panel always restores the previous search text. This PR adds an **App** setting, `app.findRestoresLastSearch` (default **on**, preserving current behavior). When disabled, find always opens with an empty search field.

## Behavior

| Scenario | Setting on (default) | Setting off |
|---|---|---|
| Cmd+F after closing a previous search | Previous text restored, fully selected | Empty field |
| Cmd+F while the find bar is already open | Unchanged (refocus, cursor preserved) | Unchanged |
| No previous search | Empty field | Empty field |

## Implementation

- The recovery decision is a single shared helper, `recoveredFindSearch()` in `Sources/App/ShortcutRoutingSupport.swift`, used by both `TabManager.startSearch()` (terminal) and `BrowserPanel.startFind()` (browser), per the shared-behavior policy. It also owns the existing select-all-on-restore rule, so the two entrypoints can't drift.
- Setting wired through: `AppCatalogSection` (`DefaultsKey`, default `true`), Settings > App row with on/off subtitles, settings sidebar navigation + search aliases + curated entry, `cmux.json` (`app.findRestoresLastSearch`: file mapping, template, supported JSON paths), and the `SettingsRowAnchorResolutionTests` contract list.

## Localization audit

New user-facing surfaces: the Settings row title and two subtitles, plus the settings-search alias. All four keys are in `Resources/Localizable.xcstrings` with `en` and `ja` entries (`settings.app.findRestoresLastSearch`, `.subtitleOn`, `.subtitleOff`, `settings.search.alias.setting.app.find-restores-last-search`). The curated search entry title uses `String(localized:)` with the same key. No web/docs message catalogs reference sibling app-behavior settings, so none needed updating.

## Testing

- New `cmuxTests/FindSearchRecoveryTests.swift` (Swift Testing, wired into `project.pbxproj`; `lint-pbxproj-test-wiring.sh` passes) covering: default restore, restore disabled, explicit enable, refocus-never-injects (both setting states), and empty-last-needle never selecting all.
- `CmuxSettings` package: 230 tests pass locally. `CmuxSettingsUI` package: 78 tests pass locally, including the updated `SettingsRowAnchorResolution` contract (94 row paths).
- App-target compile + `cmuxTests` run on CI: this machine has no zig toolchain to build GhosttyKit, so the full app build could not be validated locally. All edited app-target files pass `swiftc -parse`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/7207"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785594964&installation_id=133855650&pr_number=7207&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F7207&signature=329c0b61f2f3fb3e46a8f87dbf9a10ca48f404ce01b57e349aeae0c22f611d32"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---
## Summary by cubic
Adds a new setting to control whether Cmd+F restores the last search text. Default stays the same (restore and select-all); turning it off opens find with an empty field.

- **New Features**
  - Added `app.findRestoresLastSearch` (default true) in Settings > App; added to settings search, curated entry, path-to-anchor map, and `cmux.json`.
  - Centralized recovery logic in `FindSearchRecovery.resolve()` to keep terminal and browser behavior consistent, including select-all on restore.
  - Localized new strings and settings-search aliases across all supported locales.
  - Added `FindSearchRecoveryTests` to cover default, disabled, refocus, and empty-needle cases.

<sup>Written for commit 74a60063afa9f6751663189f1c155c6a2863c764. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/7207?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added a “Find Restores Last Search” setting to let you choose whether reopening the Find bar restores your previous query.
  * Updated Find-in-page and terminal search to follow this setting consistently.
* **Bug Fixes**
  * If restoration is disabled, reopening Find now reliably starts with an empty query.
* **Tests**
  * Added unit tests covering recovery behavior for new sessions vs refocusing the existing Find bar, including edge cases.

### CODE REVIEW

File: Packages/macOS/CmuxSettingsUI/Sources/CmuxSettingsUI/Navigation/CuratedSettingEntry+Default.swift [hunk 1]
```diff
@@ -53,6 +53,13 @@ extension Array where Element == CuratedSettingEntry {
             .init(section: .app, id: "hide-tab-close-button", title: "Hide Tab Close Button", synonyms: "app.hideTabCloseButton hide x button close tab terminal surface"),
             .init(section: .app, id: "rename-selects-name", title: "Rename Selects Existing Name", synonyms: "app.renameSelectsExistingName rename select all existing title command palette workspace name"),
             .init(section: .app, id: "palette-search-all", title: "Command Palette Searches All Surfaces", synonyms: "app.commandPaletteSearchesAllSurfaces command palette search all surfaces cmd-p terminal browser markdown"),
+            .init(
+                section: .app,
+                id: "find-restores-last-search",
+                title: String(localized: "settings.app.findRestoresLastSearch", defaultValue: "Find Restores Last Search"),
+                paths: ["app.findRestoresLastSearch"],
+                synonyms: "app.findRestoresLastSearch find restore last search previous cmd-f clear empty needle terminal browser"
```
[the remark this answers is not in the document]
Comment (prncvrm) [reply Packages/macOS/CmuxSettingsUI/Sources/CmuxSettingsUI/Navigation/CuratedSettingEntry+Default.swift:61]: The localized alias key is consumed: the app-side settings search factory merges `SettingsSearchAliasIndex.aliases(target:idSuffix:)` — which reads `settings.search.alias.setting.app.find-restores-last-search` via `localized(...)` — into the entry's searchText (`Sources/SettingsSearchAliases.swift`). The `CuratedSettingEntry` synonyms field is the package-side registry, which uses raw-English dotted-path tokens for every sibling entry (e.g. `palette-search-all`) since the package cannot reach the app's alias table; this entry follows that same pattern. The second finding (missing `settingsPathAnchorIDs` mapping) was valid and is fixed in 74a60063af.

**Comment by prncvrm** [comment]
@coderabbitai Confirmed: the cmux app target builds with SWIFT_VERSION = 5.0 and no SWIFT_DEFAULT_ACTOR_ISOLATION is set anywhere in the project, so there is no implicit MainActor-by-default isolation — `FindSearchRecovery.resolve` is already nonisolated. This is also why `startOrFocusTerminalSearch` in the same file needs its explicit `@MainActor` annotation. Leaving the type unannotated to match the file's existing style.
